cli.py

```diff
@@ -1,190 +1,95 @@
-    class Meta:
-        label = 'base'
-        description = "Cabinet's cli client for managing vaults."
-        arguments = COMMON_ARGS + [
-            (['--version'], dict(action='version', version=BANNER))
-        ]
-
-    @expose(hide=True)
-    def default(self):
-        """Without any parameter, the command will print the help"""
-        app.args.print_help()
-
-
-class ItemController(CementBaseController):
-    class Meta:
-        label = 'item'
-        stacked_on = 'base'
-        stacked_type = 'nested'
-        arguments = COMMON_ARGS + [
-            (['name'],
-             dict(help='The item name.', action='store')),
-            (['-t', '--tag'],
-             dict(help='Add a tag to the item', action='append')),
-            (['--tags'],
-             dict(help='Add multiple separated comma tags',
-                  type=tags_type, action='store')),
-            (['--content'],
-             dict(help='Add content to the item',
-                  type=tuple_type, action='append'))
-        ]
-
-    @expose(help='Get an item from the vault.')
-    def get(self):
-        """Get an item from the vault"""
-        account_id = self.app.pargs.account
-        vault_name = self.app.pargs.vault
-        name = self.app.pargs.name
-        if name:
-            self.app.log.debug('Looking for item with name "{0}"'.format(name))
-            cab = CabinetWrapper(vault_name, account_id)
-            cab.get(name)
-
-    @expose(help="Add an item to the vault.")
-    def add(self):
-        """Add an item to the vault"""
-        account_id = self.app.pargs.account
-        vault_name = self.app.pargs.vault
-        name = self.app.pargs.name
-        tags = self.app.pargs.tags
-        if not tags:
-            tags = [] if not self.app.pargs.tag else self.app.pargs.tag
-        content = [] if not self.app.pargs.content else self.app.pargs.content
-
-        content_obj = {}
-        for key, value in content:
-            content_obj[key] = value
-
-        if name:
-            cab = CabinetWrapper(vault_name, account_id)
-            cab.add_item(name, tags, content_obj)
+@click.group()
+@click.option('account', '-a', '--account',
+              help='Specify an account to use')
+@click.option('vault', '-v', '--vault',
+              help='Specify a vault to use')
+@click.pass_context
+def cli(ctx, account, vault):
+    if ctx.obj is None:
+        ctx.obj = {}
+
+    if account or vault:
+        click.echo(">> Credentials")
+        click.echo("Account:", account)
+        click.echo("Vault:", vault)
+        ctx.obj['account'] = account
+        ctx.obj['vault'] = vault
+
+
+@cli.command()
+@click.argument('name')
+@click.option('tags', '-t', '--tag', multiple=True,
+              help='Specify tags for the item')
+@click.option('content', '-c', '--content',
+              help='The item content')
+@click.option('editor', '-e', '--edit-content', is_flag=True,
+              help='The item content')
```

Comment:
yeah, it's a really nice improvement with a small amount of code :)

Agreed, I've changed the flag name and the help (which I totally forgot to change, that was a clear copy/paste grom `-c` hehe).

Yeah, it would make sense to think about using pipes to ease content editing. We may need to use a flag to say we are going to use `stdin` so we wait until the content is "entered" to ask for the password.